.github/workflows/composite-actions/kube-gateway-api-conformance-tests/action.yaml

```diff
@@ -53,6 +57,8 @@ runs:
       fi
   - name: Run the kubernetes gateway API conformance tests
     shell: bash
+    env:
+      GATEWAY_API_VERSION: ${{ inputs.gateway-api-version }}
```

Comment:
@sam-heilbron @nfuden this env var is not getting set causing the CI failure. PTAL and let me know if you see the issue or if you suggest anyone else with more familiarity with CI that can review.

Reply:
One thing that stands out to me is that we mixing "matrix" and "input" in this composite-action. I prefer your approach, since it makes the action standalone, and unaware that it's called within a matrix. I'm not sure if it causes issues combingin both, but from a mainteance perspective, I think it would be better if we adjusted the syntax to be entirely input based, as you have it